src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -122,71 +119,433 @@ namespace GroundHeatExchangers {
-    std::vector<std::shared_ptr<GLHEVertPropsStruct>> vertPropsVector;
-    std::vector<std::shared_ptr<GLHEResponseFactorsStruct>> responseFactorsVector;
-    std::vector<std::shared_ptr<GLHEVertSingleStruct>> singleBoreholesVector;
+    constexpr Real64 hrsPerMonth(730.0); // Number of hours in month
+    constexpr Real64 maxTSinHr(60);      // Max number of time step in a hour
+
+    //******************************************************************************
+
+    GLHESlinky::GLHESlinky(EnergyPlusData &state, std::string const &objName, nlohmann::json const &j)
+    {
+        // Check for duplicates
+        for (auto &existingObj : state.dataGroundHeatExchanger->singleBoreholesVector) {
+            if (objName == existingObj->name) {
+                ShowFatalError(state, "Invalid input for " + this->moduleName + " object: Duplicate name found: " + existingObj->name);
+            }
+        }
+
+        bool errorsFound = false;
+
+        this->name = objName;
+
+        std::string inletNodeName = UtilityRoutines::MakeUPPERCase(j["inlet_node_name"]);
+        std::string outletNodeName = UtilityRoutines::MakeUPPERCase(j["outlet_node_name"]);
+
+        // get inlet node num
+        this->inletNodeNum = NodeInputManager::GetOnlySingleNode(
+            state, inletNodeName, errorsFound, this->moduleName, this->name, NodeType_Water, NodeConnectionType_Inlet, 1, ObjectIsNotParent);
+
+        // get outlet node num
+        this->outletNodeNum = NodeInputManager::GetOnlySingleNode(
+            state, outletNodeName, errorsFound, this->moduleName, this->name, NodeType_Water, NodeConnectionType_Outlet, 1, ObjectIsNotParent);
+
+        this->available = true;
+        this->on = true;
+
+        BranchNodeConnections::TestCompSet(state, this->moduleName, this->name, inletNodeName, outletNodeName, "Condenser Water Nodes");
+
+        // load data
+        this->designFlow = j["design_flow_rate"];
+        PlantUtilities::RegisterPlantCompDesignFlow(this->inletNodeNum, this->designFlow);
+
+        this->soil.k = j["soil_thermal_conductivity"];
+
+        Real64 const rho = j["soil_density"];
+        Real64 const cp = j["soil_specific_heat"];
+        this->soil.rhoCp = rho * cp;
+        this->pipe.k = j["pipe_thermal_conductivity"];
+        this->pipe.rho = j["pipe_density"];
+        this->pipe.cp = j["pipe_specific_heat"];
+        this->pipe.outDia = j["pipe_outer_diameter"];
+        this->pipe.outRadius = this->pipe.outDia / 2.0;
+        this->pipe.thickness = j["pipe_thickness"];
+
+        std::string const hxConfig = UtilityRoutines::MakeUPPERCase(j["heat_exchanger_configuration"]);
+        if (UtilityRoutines::SameString(hxConfig, "VERTICAL")) {
+            this->verticalConfig = true;
+        } else if (UtilityRoutines::SameString(hxConfig, "HORIZONTAL")) {
+            this->verticalConfig = false;
+        }
+
+        this->coilDiameter = j["coil_diameter"];
```

Comment:
@mitchute This is an interesting direction to move in. 
1. What happens if the field name string is misspelled? Will it throw an error or simply return a zero or blank?
2. What happens if the field isn't present in the incoming object? Will it be present in the object data with it's default if it has one?
I still feel that we need an input fetching utility to manage this as suggested in https://github.com/NREL/EnergyPlus/issues/7632 .
Or maybe fields with defaults are already populated in the JSON data structure when the input is read in? I've never been sure about that.

Reply:
@mjwitte

Re 1. I just tested this by misspelling one field, ```depth_of_top_of_borehole```. The following is the error message thrown.

```
** Severe  ** [json.exception.type_error.302] type must be number, but is number
```

Not super helpful, but one would hope that these sort of errors never make it past the developer/reviewer since the file likely would never run. We should give this a little more thought.

Re 2. We handle fields that are could be missing and which do NOT have default values like the following:
```C++
if (j.find("ghe_vertical_responsefactors_object_name") != j.end())
```
In this case, the field will not be present if it's not present in the input. I'd assume that fields with default values will have the default filled in automatically, but I have not tested that.

FFR, some issues I had with this transition are listed here:
 - The string fields come in as-is from the input. I found myself needing to upper case everything before comparing strings, whereas before I think some of this was done automatically. We'll need a way to keeps some fields case sensitive, such as paths. However, the majority of strings don't need that, so when we go to scale this up we'll need to decide how to best handle it.
 - Currently, we have ```cAlphaFieldNames``` and ```cNumericFieldNames``` which allows us to programmatically get the field names. I didn't figure out a way to get these using JSON, but it probably exists. Either way, the names are likely to come in as the JSON key names. We may want a way to provide the user with the IDF-like field names, i.e. "inlet_node_name" vs. "Inlet Node Name".

This was my first real attempt at using the JSON inputs, so perhaps these issues were just due to my inexperience.